x-pack/test/epm_api_integration/apis/data_sources.ts

```diff
@@ -86,6 +86,18 @@ export default function({ getService }: FtrProviderContext) {
 
       const readDataSourceSavedObject = async () => {
         const response = await supertest
+          // I tried changing this to
+          // /api/saved_objects/datasources/yamlpipeline-1.0.0
+          // b/c `datasources` is the name ingest uses but it 404'd
+          // /api/saved_objects/_find?type=datasources
+          // and
+          // /api/ingest/datasources
+          // both show the saved object
+          // I tried adding
+          // datasources: { isNamespaceAgnostic: true, }
+          // to https://github.com/elastic/kibana/blob/ef9bc478cba32eb8722c17d7911cb201941f2adc/x-pack/legacy/plugins/ingest/index.ts#L37
+          // thinking that's what registered it as a type but it didn't work
+          // I didn't do a full restart though, so maybe it does still work
           .get('/api/saved_objects/epm-datasource/yamlpipeline-1.0.0')
```

Comment:
I just realized this should test Ingest's API; not the layer underneath. Our contract is the API; they can move / update and EPM shouldn't care.

However, not sure if "get by id" is working

```
‣ curl --user elastic:changeme localhost:5601/api/ingest/datasources
{"list":[],"success":true,"total":0,"page":1,"perPage":20}
‣ curl --user elastic:changeme localhost:5601/api/epm/install/yamlpipeline-1.0.0
[]
‣ curl --user elastic:changeme localhost:5601/api/epm/datasource/install/yamlpipeline-1.0.0
[
  {
    "id": "yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_json_yml",
    "type": "ingest-pipeline"
  },
  {
    "id": "yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_entry_yml",
    "type": "ingest-pipeline"
  },
  {
    "id": "yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_plaintext_yml",
    "type": "ingest-pipeline"
  }
]
‣ curl --user elastic:changeme localhost:5601/api/ingest/datasources
{"list":[{"id":"yamlpipeline-1.0.0","name":"name","read_alias":"read_alias","package":{"name":"yamlpipeline","version":"1.0.0","description":"This package contains a yaml pipeline.\n","title":"Yaml Pipeline package","assets":[{"id":"yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_json_yml","type":"ingest-pipeline"},{"id":"yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_entry_yml","type":"ingest-pipeline"},{"id":"yamlpipeline_1_0_0_dataset_log_elasticsearch_ingest_pipeline_pipeline_plaintext_yml","type":"ingest-pipeline"}]},"streams":[{"id":"string","input":{"type":"log","config":{"config":"values","go":"here"},"ingest_pipelines":["string"],"id":"string","index_template":"string","ilm_policy":"string","fields":[{}]},"config":{"config":"values","go":"here"},"output_id":"output_id","processors":["string"]}]}],"success":true,"total":1,"page":1,"perPage":20}
‣ curl --user elastic:changeme localhost:5601/api/ingest/datasources/yamlpipeline-1.0.0
{"item":null,"success":true}
```

The datasource shows up in the list, but getting by id returns null

Reply:
> I just realized this should test Ingest's API; not the layer underneath.

In theory, I totally agree. In practice, I think we need to test everything here that can't / isn't be tested elsewhere. When I wrote this test, this included checking that the saved object is saved correctly, as it is part of EPM and we don't test that elsewhere.

More important than checking the saved object here will be tests that ensure that the generation and installation of datasource-specific assets (index templates, ILM policies, ingest pipelines, possibly more) is working correctly.

Reply:
I guess my comment would have been more clear if I wrote "should now" or something. I'm just saying that since the rules have changed (EPM is no longer saving the object) the test should change. We'll be using the Ingest API so we should confirm that it behaves as we expect.

Totally agree about tests for the assets/other side effects.